### PR TITLE
Refactor API request to include Content-Type header

### PR DESCRIPTION
**Pull request by Yongtae723**
this is copy from [discord](https://discord.com/channels/1087698854775881778/1096265567058534461/1130069743760588883)

after I deploy flowise on render.com and make some LLM chain, we can not call the LLM via curl and javascript, but python can connect.
after some workaround, I found we have to explicitly specify the encoding of data in JSON, for example adding -H "Content-Type: application/json"in curl command.
python do this specification and thus no error happened.

this is jacascript version

### CODE REVIEW

File: packages/ui/src/views/chatflows/APICodeDialog.js [hunk 1]
```diff
@@ -303,7 +309,10 @@ async function query(formData) {
         "${baseURL}/api/v1/prediction/${dialogProps.chatflowid}",
         {
             method: "POST",
-            body: formData
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify(formData)
```
Comment (HenryHengZJ) [inline packages/ui/src/views/chatflows/APICodeDialog.js:315]: Dont think this works. Using `formData` , the header should be `'Content-Type': 'application/x-www-form-urlencoded'` not `application/json`, so I prefer to leave this out.

You also cant `JSON stringify` formData

Comment (Yongtae723) [reply packages/ui/src/views/chatflows/APICodeDialog.js]: oh sorry. you are correct.

thank you for pointing out this point!

File: packages/ui/src/views/chatflows/APICodeDialog.js [hunk 2]
```diff
@@ -347,9 +356,12 @@ async function query(formData) {
...
         "${baseURL}/api/v1/prediction/${dialogProps.chatflowid}",
         {
-            headers: { Authorization: "Bearer ${selectedApiKey?.apiKey}" },
+            headers: {
+                Authorization: "Bearer ${selectedApiKey?.apiKey}",
+                "Content-Type": "application/x-www-form-urlencoded"
+            },
             method: "POST",
-            body: formData
+            body: new URLSearchParams(formData)
```
Comment (HenryHengZJ) [inline packages/ui/src/views/chatflows/APICodeDialog.js:364]: do we need `new URLSearchParams` ? 

actually i dont think we should set `"Content-Type": "application/x-www-form-urlencoded"` according to many [answers](https://stackoverflow.com/questions/62644988/using-formdata-inside-fetch-api-is-not-working) online, just let the fetch decide for you

Comment (Yongtae723) [reply packages/ui/src/views/chatflows/APICodeDialog.js:364]: I see.

then I'll revert to the changes related to the form.

**Review by HenryHengZJ — APPROVED** [review]
thank you @Yongtae723 !

**Pull request merged**